### ISSUE TITLE
NEW: Check failed: status == CUBLAS STATUS_SUCCESS (11 vs. 0) CUBLAS STATUS MAPPING_ERROR

### ISSUE THREAD

**Issue by isn4** [alexgkendall/caffe-segnet#101]
Hello, like many, I ran into the "Check failed: status == CUBLAS STATUS_SUCCESS (11 vs. 0) CUBLAS STATUS MAPPING_ERROR" error.
I've looked around at other issues submitted similar to this one and can only find the advice that I set num_output from my conv1_1_D layer to 2 and that I make sure my softmax with loss layer has only 2 class weightings. Additionally, my images are the same size (1, 3, 360, 480) and are of the same count as the original SegNet data (367). I am also using 3 channel rgb images and 1 channel greyscale labels.
I initially thought the issue was that I cannot truly force PNG files to have only a single channel as the alpha channel (even when off) still exists.
Thus I converted my PNG files into grayscale JPG files and tried the net with those but because JPGs use compression, the pixel values in my images always have more classes (ie, more variations in the pixel values rather than just 0 and 255 as I would like and as they were in the PNG files).
@alexgkendall How were you able to get your PNG files in the original segnet to pass through segnet with their implicit alpha channels? Any conversion to grayscale I make "fakes" grayscale and gives me sRGB files with "turned off" alpha channels that still exist.
The last piece of advice I was given was to delete my previous snapshot files then re-run the current model I'm working with. I returned to using the PNG files (because of the class number issues I faced with the JPGs) and deleted the previous snapshot files and re-ran the model but am still facing this issue.

I've followed all of the advice I could find but am still running into this issue. My net is unchanged from the original SegNet version save for the conv1_1_D layer and the softmax with loss layer:

layer {
bottom: "conv1_2_D"
top: "conv1_1_D"
name: "conv1_1_D"
type: "Convolution"
param {
lr_mult: 1
decay_mult: 1
}
param {
lr_mult: 2
decay_mult: 0
}
convolution_param {
weight_filler {
type: "msra"
}
bias_filler {
type: "constant"
}
num_output: 2
pad: 1
kernel_size: 3
}
}
layer {
name: "loss"
type: "SoftmaxWithLoss"
bottom: "conv1_1_D"
bottom: "label"
top: "loss"
softmax_param {engine: CAFFE}
loss_param: {
weight_by_label_freqs: true
class_weighting: 0
class_weighting: 1
}
}

I would so appreciate any help!
Thank you!

**Comment by p-kleczek** [comment]
@isn4 Well, my optimism was premature - I managed to run my network twice and then the error appeared again.  I didn't change any data - neither in my network model nor in my training set.

**Comment by eliethesaiyan** [comment]
@alexgkendall how did you convert the true label to have single channel grayscale  with the pixel values as the label?

**Comment by isn4** [comment]
FIXED: Turns out you have to change the range of the pixel values as well as the actual number of pixel values. Segnet gets confused if you have 256 possible pixel values (0-255) and don't have class weightings for each of them. So I changed all of my PNG label images from 255 and 0 as the pixel possibilities to 1 and 0 as the pixel possibilities.
Here's my python script for doing so:
`import os`
`import cv2`
`import numpy as np`
`img = cv2.imread('/usr/local/project/old_png_labels/label.png, 0)`
`a_img = np.array(img, np.double)`
`normalized = cv2.normalize(img, a_img, 1.0, 0.0, cv2.NORM_MINMAX)`
`cv2.imwrite('/usr/local//project/png_labels/label.png, normalized)`

**Comment by ShuailingZhao** [comment]
I set the last convolution layer num_output to 2, and got the same mistake with weight_by_label_freqs being false. When I set the weight_by_label_freqs to true  and remained two class_weightings, it worked, just likt this:
layer {
  bottom: "conv1_2_D"
  top: "conv1_1_D"
  name: "conv1_1_D"
  type: "Convolution"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param { 
    weight_filler {
      type: "msra"
    }
    bias_filler {
      type: "constant"
    }
    num_output: 2
    pad: 1
    kernel_size: 3
  }
}
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "conv1_1_D"
  bottom: "label"
  top: "loss"
  softmax_param {engine: CAFFE}
  loss_param: {
    weight_by_label_freqs: true
    ignore_label: 2
    class_weighting: 1.5169
    class_weighting: 0.7458
  }
}
Maybe you can try, but I donnot understand Why exactly.

**Comment by isn4** [comment]
Hi @ShuailingZhao - absolutely. I had the weight_by_label_freqs set to true and had 2 class weightings in the net. The issue was that my label images had pixel values in the range of 0-255. The net gets confused by that because it "sees" 256 classes in label images like that. So I had to set the label image pixel value range to 0-1 that way the net wasn't confused.

**Comment by coderzbx** [comment]
yeah,this answer works! @isn4

**Comment by root1369** [comment]
I'm training my own dataset with 2 classes. Converted labels using the python script shared by @isn4 (thanks a lot!), provided class label from a text file and set color map from label images. While model creation used the original FCN Alexnet Prototxt that comes with the PASCAL VOC dataset. 
After 2 epochs aborted the training as I wanted to infer my model and see if everything is working properly. Followed inference method mentioned here for the PASCAL VOC dataset: https://github.com/NVIDIA/DIGITS/tree/master/examples/semantic-segmentation
But while inference error shows "No palette found in dataset - choose other colormap"
Then I set "colormap" in Visualization Options as "None(Grayscale)" but I get the same image in inference. Need guidance on this. Thanks!
![train_graph](https://user-images.githubusercontent.com/43385355/49646205-6480dc00-fa59-11e8-99d7-9bc307a1555e.png)
![inference_method](https://user-images.githubusercontent.com/43385355/49646219-6ea2da80-fa59-11e8-9364-cbff5996591a.png)
![inference_result](https://user-images.githubusercontent.com/43385355/49646221-74002500-fa59-11e8-9c03-a85067e04503.png)
